src/meshapi/tasks.py

```diff
@@ -12,21 +13,61 @@
 )
 from meshapi.views.panoramas import sync_github_panoramas
 from meshdb.celery import app as celery_app
+from meshdb.settings import DBBACKUP_STORAGE_OPTIONS
 
 
 @celery_app.task
-def run_database_backup() -> None:
-    logging.info("Running database backup task")
+def run_database_backup() -> bool:
+    # Don't run a backup unless it's prod1
+    environment = f"\"{os.environ.get('MESHDB_ENVIRONMENT', '')}\""
+    if os.environ.get("MESHDB_ENVIRONMENT") != "prod1":
+        logging.warn(f"Not running database backup. This environment is: {environment}")
+        return False
+
+    logging.info(f"Running database backup task. This environment is {environment}")
     if not os.environ.get("AWS_ACCESS_KEY_ID") or not os.environ.get("AWS_SECRET_ACCESS_KEY"):
         logging.error("Could not run backup. Missing AWS credentials!")
-        return
+        return False
 
     try:
         management.call_command("dbbackup")
     except Exception as e:
         logging.exception(e)
         raise e
 
+    return True
+
+
+@celery_app.task
+def reset_dev_database() -> bool:
+    # Only reset dev environments (very important!)
+    environment = f"\"{os.environ.get('MESHDB_ENVIRONMENT', '')}\""
+    if "dev" not in os.environ.get("MESHDB_ENVIRONMENT", ""):
+        logging.warn(f"Not resetting this database. This environment is: {environment}")
+        return False
+
+    logging.info(f"Running database reset task. This environment is: {environment}")
+
+    if not os.environ.get("AWS_ACCESS_KEY_ID") or not os.environ.get("AWS_SECRET_ACCESS_KEY"):
+        logging.error("Could not run backup. Missing AWS credentials!")
+        return False
+
+    try:
+        latest_backup = task_utils.get_most_recent_object(
+            DBBACKUP_STORAGE_OPTIONS["bucket_name"], DBBACKUP_STORAGE_OPTIONS["location"]
+        )
+
+        if not latest_backup:
+            raise ValueError("Could not get most recent dbbackup while resetting dev environment.")
+
+        management.call_command("dbrestore", "--noinput", "-i", latest_backup)
+        management.call_command("scramble_members", "--noinput")
```

Comment:
Do we need to consider the security implications of a bad guy having dev access and getting de-anonymized data in between when these commands run?

Reply:
Hmmm, I wonder how practical it would be to have some kind of sandbox to do the scramble in, or swap between a primary and secondary database. It would add some complexity... Alternatively we can maybe kick everyone out while this job runs.

Reply:
Yeah that may be a simpler solution. I don't really like the complexity of having to hot-swap postgres databases, but I bet django supports it to at least some degree

Reply:
Yeah, I would like to not do all that nonsense just to ensure security. I might look into logging people out.

Reply:
Even that may be trickier than you expect, since you’ll need to invalidate all existing sessions, as well as prevent the creation of new ones

Reply:
Ah, I can create maintenance mode middleware: https://medium.com/@keagileageek/putting-a-django-site-in-maintenance-mode-creating-a-django-maintenance-mode-middleware-7c5474f62491

Reply:
I have done it

but at what cost?